public/components/trace_analytics/requests/request_handler.ts

```diff
@@ -25,35 +31,45 @@ export function handleDslRequest(http: CoreStart['http'], DSL: any, bodyQuery: a
   }
   let body = bodyQuery;
   if (!bodyQuery.index) {
-    body = {...bodyQuery, index: ((mode === 'jaeger' ? JAEGER_INDEX_NAME : DATA_PREPPER_INDEX_NAME)) }
+    body = { ...bodyQuery, index: mode === 'jaeger' ? JAEGER_INDEX_NAME : DATA_PREPPER_INDEX_NAME };
   }
   if (timeout) {
     const id = setTimeout(() => setShowTimeoutToast!(), 30000);
-    return http
-      .post(TRACE_ANALYTICS_DSL_ROUTE, {
+
+    try {
+      return await http.post(TRACE_ANALYTICS_DSL_ROUTE, {
         body: JSON.stringify(body),
-      })
-      .catch((error) => {
-        console.error(error)
-      })
-      .finally(() => clearTimeout(id));
+      });
+    } catch (error) {
+      console.error(error);
+    } finally {
+      clearTimeout(id);
+    }
   }
 
-  return http
-    .post(TRACE_ANALYTICS_DSL_ROUTE, {
+  try {
+    return await http.post(TRACE_ANALYTICS_DSL_ROUTE, {
       body: JSON.stringify(body),
-    })
-    .catch((error) => console.error(error));
+    });
+  } catch (error_1) {
+    console.error(error_1);
+  }
```

Comment:
what's the difference?

Reply:
There's no difference in logic/code. The function is converted to an Async call. Without this we run into this failure: https://github.com/opensearch-project/dashboards-observability/actions/runs/5092646146/jobs/9158330893?pr=489